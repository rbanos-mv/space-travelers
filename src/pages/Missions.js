const Missions = () => (
  <div>
    Missions page
  </div>
);

export default Missions;
