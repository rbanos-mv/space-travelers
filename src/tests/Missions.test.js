import { fireEvent, render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import Missions from '../pages/Missions';
import setupStore from './setupStore';

describe('Test the Missions components', () => {
  it('renders the Missions page correctly', () => {
    const store = setupStore(false);
    const page = render(
      <Provider store={store}>
        <Missions />
      </Provider>,
    );
    expect(page).toMatchSnapshot();
  });

  it('shows the button changes from "Join Mission" to "Leave Mission" on click event', () => {
    const store = setupStore(false);
    render(
      <Provider store={store}>
        <Missions />
      </Provider>,
    );

    expect(screen.queryByText('Leave Mission')).toBeFalsy();
    fireEvent.click(screen.queryByText('Join Mission'));
    expect(screen.queryByText('Leave Mission')).toBeTruthy();
    expect(screen.queryByText('Join Mission')).toBeFalsy();
  });

  it('shows the button changes from "Leave Mission" to "Join Mission" on click event', () => {
    const store = setupStore(true);
    render(
      <Provider store={store}>
        <Missions />
      </Provider>,
    );

    expect(screen.queryByText('Join Mission')).toBeFalsy();
    fireEvent.click(screen.queryByText('Leave Mission'));
    expect(screen.queryByText('Join Mission')).toBeTruthy();
    expect(screen.queryByText('Leave Mission')).toBeFalsy();
  });

  it('shows the active member badge on click event', () => {
    const store = setupStore(false);
    render(
      <Provider store={store}>
        <Missions />
      </Provider>,
    );

    expect(screen.queryByText('Active Member')).toBeFalsy();
    expect(screen.queryByText('NOT A MEMBER')).toBeTruthy();
    fireEvent.click(screen.queryByText('Join Mission'));
    expect(screen.queryByText('Active Member')).toBeTruthy();
    expect(screen.queryByText('NOT A MEMBER')).toBeFalsy();
  });

  it('hides the active member badge on click event', () => {
    const store = setupStore(true);
    render(
      <Provider store={store}>
        <Missions />
      </Provider>,
    );

    expect(screen.queryByText('Active Member')).toBeTruthy();
    expect(screen.queryByText('NOT A MEMBER')).toBeFalsy();
    fireEvent.click(screen.queryByText('Leave Mission'));
    expect(screen.queryByText('Active Member')).toBeFalsy();
    expect(screen.queryByText('NOT A MEMBER')).toBeTruthy();
  });
});
