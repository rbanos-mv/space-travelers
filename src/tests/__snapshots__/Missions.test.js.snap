// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Missions components renders the Missions page correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="missions-page"
      >
        <table>
          <thead>
            <tr>
              <th>
                <h4
                  class="mission-title"
                >
                  Mission
                </h4>
              </th>
              <th
                class="desc-col"
              >
                <p>
                  Description
                </p>
              </th>
              <th
                class="status-col"
              >
                <p>
                  Status
                </p>
              </th>
              <th
                class="join-col"
              >
                 
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td
                class="left-col"
              >
                Thaicom
              </td>
              <td>
                Thaicom is the name of a series of communications satellites operated from Thailand, and also the name of Thaicom Public Company Limited, which is the company that owns and operates the Thaicom satellite fleet and other telecommunication businesses in Thailand and throughout the Asia-Pacific region. The satellite projects were named Thaicom by the King of Thailand, His Majesty the King Bhumibol Adulyadej, as a symbol of the linkage between Thailand and modern communications technology.
              </td>
              <td>
                <div
                  class="status"
                >
                  NOT A MEMBER
                </div>
              </td>
              <td>
                <button
                  class="join-btn"
                  type="button"
                >
                  Join Mission
                </button>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="missions-page"
    >
      <table>
        <thead>
          <tr>
            <th>
              <h4
                class="mission-title"
              >
                Mission
              </h4>
            </th>
            <th
              class="desc-col"
            >
              <p>
                Description
              </p>
            </th>
            <th
              class="status-col"
            >
              <p>
                Status
              </p>
            </th>
            <th
              class="join-col"
            >
               
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td
              class="left-col"
            >
              Thaicom
            </td>
            <td>
              Thaicom is the name of a series of communications satellites operated from Thailand, and also the name of Thaicom Public Company Limited, which is the company that owns and operates the Thaicom satellite fleet and other telecommunication businesses in Thailand and throughout the Asia-Pacific region. The satellite projects were named Thaicom by the King of Thailand, His Majesty the King Bhumibol Adulyadej, as a symbol of the linkage between Thailand and modern communications technology.
            </td>
            <td>
              <div
                class="status"
              >
                NOT A MEMBER
              </div>
            </td>
            <td>
              <button
                class="join-btn"
                type="button"
              >
                Join Mission
              </button>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
